lib/python/bind_units.cpp

```diff
@@ -5,16 +5,77 @@
 #include "scipp/core/dtype.h"
 #include "scipp/core/tag_util.h"
 #include "scipp/units/unit.h"
-#include "scipp/variable/variable.h"
 
-#include "dtype.h"
-#include "numpy.h"
 #include "pybind11.h"
 #include "unit.h"
 
 using namespace scipp;
 namespace py = pybind11;
 
+constexpr int UNIT_DICT_VERSION = 1;
+
+py::dict to_dict(const units::Unit &unit) {
+  py::dict dict;
+  dict["__version__"] = UNIT_DICT_VERSION;
+
+  const auto &base_units = unit.underlying().base_units();
+  dict["meter"] = base_units.meter();
+  dict["kilogram"] = base_units.kg();
+  dict["second"] = base_units.second();
+  dict["ampere"] = base_units.ampere();
+  dict["kelvin"] = base_units.kelvin();
+  dict["mole"] = base_units.mole();
+  dict["candela"] = base_units.candela();
+  dict["currency"] = base_units.currency();
+  dict["count"] = base_units.count();
+  dict["radian"] = base_units.radian();
+  // Returning ints instead of bools because
+  // - The constructor of unit_data takes unsigned int as arguments.
+  // - h5py saves bools as enum types which waste disk space.
+  dict["per_unit"] = static_cast<unsigned int>(base_units.is_per_unit());
+  dict["i_flag"] = static_cast<unsigned int>(base_units.has_i_flag());
+  dict["e_flag"] = static_cast<unsigned int>(base_units.has_e_flag());
+  dict["equation"] = static_cast<unsigned int>(base_units.is_equation());
+
+  // We loose some type information. commodity is uint32
+  // but the dict contains a signed int.
+  // This should not matter because Python's int is larger than 32 bit.
+  dict["commodity"] = unit.underlying().commodity();
```

Comment:
We are not using/supporting this either, are we?

Reply:
No we are not. We can remove it. But do we gain much from that given that the rest is tied closely to the internals of LLNL/Units?

Reply:
I am not concerned tying this code to LLNL/units, but about tying our *files* to it. Breaking the file format if we decided to switch may be a big problem in the future.